app/uk/gov/hmrc/agentsubscriptionfrontend/models/AddressValidator.scala

```diff
@@ -28,74 +28,83 @@ import scala.util.matching.Regex
 
 object AddressValidator {
 
-  type ValidationErrors = NonEmptyList[ValidationError]
+  type ValidatedType[A] = ValidatedNel[ValidationError, A]
```

Comment:
I think that `ValidatedType` makes things less clear - I think it should be inlined